beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -226,6 +228,43 @@ func (s *Service) validateBeaconBlock(ctx context.Context, blk block.SignedBeaco
 		return errors.New("incorrect proposer index")
 	}
 
+	return validateBellatrixBeaconBlock(parentState, blk.Block())
+}
+
+// validateBellatrixBeaconBlock validates the block for the Bellatrix fork.
+// spec code:
+//   If the execution is enabled for the block -- i.e. is_execution_enabled(state, block.body) then validate the following:
+//      [REJECT] The block's execution payload timestamp is correct with respect to the slot --
+//      i.e. execution_payload.timestamp == compute_timestamp_at_slot(state, block.slot).
+func validateBellatrixBeaconBlock(parentState state.BeaconState, blk block.BeaconBlock) error {
+	if parentState.Version() != version.Bellatrix || blk.Version() != version.Bellatrix {
```

Comment:
> The parent state will be an altair state, so this block will skip all the relevant execution payload checks.

I don't think this is true. The parent state should already be upgraded to Bellatrix state via:
```go
if features.Get().EnableNextSlotStateCache {
		parentState, err = transition.ProcessSlotsUsingNextSlotCache(ctx, parentState, blk.Block().ParentRoot(), blk.Block().Slot())
		if err != nil {
			return err
		}
	} else {
		parentState, err = transition.ProcessSlots(ctx, parentState, blk.Block().Slot())
		if err != nil {
			return err
		}
	}
```

> this will allow a non bellatrix block that references a bellatrix parent state to be valid. It won't return an error and is passed on towards the subscriber.

true, although unlikely, we certainly can add checks around it